consensus/spos/consensusMessageValidator.go

```diff
@@ -0,0 +1,270 @@
+			ErrInvalidBodySize,
+			len(cnsMsg.Body))
+	}
+
+	if cnsMsg.Header == nil || len(cnsMsg.Header) > core.MegabyteSize {
+		return fmt.Errorf("%w : received header from consensus topic has an invalid size: %d",
+			ErrInvalidHeaderSize,
+			len(cnsMsg.Header))
+	}
+
+	return nil
+}
+
+func (wrk *Worker) checkMessageWithBlockBodyValidity(cnsMsg *consensus.Message) error {
+	isMessageInvalid := cnsMsg.Header != nil ||
+		cnsMsg.SignatureShare != nil ||
+		cnsMsg.PubKeysBitmap != nil ||
+		cnsMsg.AggregateSignature != nil ||
+		cnsMsg.LeaderSignature != nil
+
+	if isMessageInvalid {
+		return fmt.Errorf("%w : received message from consensus topic is invalid: "+
+			"len(Header) = %d SignatureShare = %s PubKeysBitmap = %s AggregateSignature = %s LeaderSignature = %s",
+			ErrInvalidMessage,
+			len(cnsMsg.Header),
+			logger.DisplayByteSlice(cnsMsg.SignatureShare),
+			logger.DisplayByteSlice(cnsMsg.PubKeysBitmap),
+			logger.DisplayByteSlice(cnsMsg.AggregateSignature),
+			logger.DisplayByteSlice(cnsMsg.LeaderSignature))
+	}
+
+	if len(cnsMsg.Body) > core.MegabyteSize {
+		return fmt.Errorf("%w : received body from consensus topic has an invalid size: %d",
+			ErrInvalidBodySize,
+			len(cnsMsg.Body))
+	}
+
+	return nil
+}
+
+func (wrk *Worker) checkMessageWithBlockHeaderValidity(cnsMsg *consensus.Message) error {
+	isMessageInvalid := cnsMsg.Body != nil ||
+		cnsMsg.SignatureShare != nil ||
+		cnsMsg.PubKeysBitmap != nil ||
+		cnsMsg.AggregateSignature != nil ||
+		cnsMsg.LeaderSignature != nil
+
+	if isMessageInvalid {
+		return fmt.Errorf("%w : received message from consensus topic is invalid: "+
+			"len(Body) = %d SignatureShare = %s PubKeysBitmap = %s AggregateSignature = %s LeaderSignature = %s",
+			ErrInvalidMessage,
+			len(cnsMsg.Body),
+			logger.DisplayByteSlice(cnsMsg.SignatureShare),
+			logger.DisplayByteSlice(cnsMsg.PubKeysBitmap),
+			logger.DisplayByteSlice(cnsMsg.AggregateSignature),
+			logger.DisplayByteSlice(cnsMsg.LeaderSignature))
+	}
+
+	if cnsMsg.Header == nil || len(cnsMsg.Header) > core.MegabyteSize {
+		return fmt.Errorf("%w : received header from consensus topic has an invalid size: %d",
+			ErrInvalidHeaderSize,
+			len(cnsMsg.Header))
+	}
+
+	return nil
+}
+
+func (wrk *Worker) checkMessageWithSignatureValidity(cnsMsg *consensus.Message) error {
+	isMessageInvalid := cnsMsg.Body != nil ||
+		cnsMsg.Header != nil ||
+		cnsMsg.PubKeysBitmap != nil ||
+		cnsMsg.AggregateSignature != nil ||
+		cnsMsg.LeaderSignature != nil
+
+	if isMessageInvalid {
+		return fmt.Errorf("%w : received message from consensus topic is invalid: "+
+			"len(Body) = %d len(Header) = %d PubKeysBitmap = %s AggregateSignature = %s LeaderSignature = %s",
+			ErrInvalidMessage,
+			len(cnsMsg.Body),
+			len(cnsMsg.Header),
+			logger.DisplayByteSlice(cnsMsg.PubKeysBitmap),
+			logger.DisplayByteSlice(cnsMsg.AggregateSignature),
+			logger.DisplayByteSlice(cnsMsg.LeaderSignature))
+	}
+
+	if len(cnsMsg.SignatureShare) != core.SignatureSizeInBytes {
```

Comment:
we should not have the const core.SignatureSizeInBytes as it defeats the purpose of changing the signature type from config.

Reply:
I would pull it in config.toml in [Consensus] section

Reply:
I should have made it more clear:
HashSizeInBytes is not required, as Hasher interface has the Size() method
PublicKeySizeInBytes can be recovered from the config, from the Address and BLSPublicKey area. 
SignatureSizeInBytes might be included in the aforementioned Address and BLSPublicKey as to be changed on the same time.

Reply:
Done